lmdk/cmake/xtensa.cmake

```diff
@@ -0,0 +1,25 @@
+# Xtensa CMake toolchain file. Apply it using CMAKE_TOOLCHAIN_FILE variable.
+
+set(CMAKE_SYSTEM_NAME Generic)
+set(CMAKE_SYSTEM_VERSION 1)
+
+cmake_path(CONVERT $ENV{XTENSA_TOOLCHAIN_PATH} TO_CMAKE_PATH_LIST XTENSA_TOOLCHAIN_PATH)
+set(TOOLCHAIN_BASE ${XTENSA_TOOLCHAIN_PATH}/XtensaTools)
+set(CROSS_COMPILE ${TOOLCHAIN_BASE}/bin/xt-)
+
+# clang or xcc
+find_program(CMAKE_C_COMPILER NAMES ${CROSS_COMPILE}xcc NO_DEFAULT_PATH)
+# clang++ or xc++
+find_program(CMAKE_CXX_COMPILER NAMES ${CROSS_COMPILE}xc++ NO_DEFAULT_PATH)
```

Comment:
GCC should also work here.

Reply:
Renamed toolchain file to xtensa-toolchain.cmake. The file is not included but used as
```
set(CMAKE_TOOLCHAIN_FILE "${CMAKE_CURRENT_LIST_DIR}/../../cmake/xtensa-toolchain.cmake")
```
in dummy library CMakeLists.txt. So if that line is commented out, the toolchain file may be used with cmake --tollchain=<path> option.